src/components/EarningsTable/EarningsTable.js

```diff
@@ -123,28 +124,43 @@ function EarningsTable() {
   const [selectedEarning, setSelectedEarning] = useState(null);
 
   async function getEarnings(fetchAll = false) {
-    console.warn('getEarnings with fetchAll: ', fetchAll);
+    // console.warn('getEarnings with fetchAll: ', fetchAll);
     setIsLoading(true); // show loading indicator when fetching data
 
-    const { results, totalCount } = await getEarningsReal();
+    const { results, totalCount } = await getEarningsReal(fetchAll);
     setEarnings(results);
     setTotalEarnings(totalCount);
 
     setIsLoading(false); // hide loading indicator when data is fetched
   }
 
   async function getEarningsReal(fetchAll = false) {
-    console.warn('fetchAll:', fetchAll);
+    // console.warn('fetchAll:', fetchAll);
+    const filtersToSubmit = { ...filter };
+    // filter out keys we don't want to submit
+    Object.keys(filtersToSubmit).forEach((k) => {
+      if (
+        filtersToSubmit[k] === 'all' ||
```

Comment:
This is a risky way of spotting that all values should be included.
If a user wants to filter by a grower name that starts with or contains "all", we will ignore the value and return all growers.

Reply:
I see that it's just been moved from another part of the code, but we should try to resolve it.

Reply:
I'll make another issue